src/common/k8s-api/kube-api-parse.ts

```diff
@@ -8,11 +8,11 @@
 import { splitArray } from "../utils";
 
 export interface IKubeApiLinkRef {
-  apiPrefix?: string;
+  apiPrefix?: string | undefined;
   apiVersion: string;
   resource: string;
-  name?: string;
-  namespace?: string;
+  name?: string | undefined;
```

Comment:
There's already `?` making the '| undefined' add weird noise. There's a lot of changes like this. I wonder if this specific new kind of strictness in typing is worth it. I do not understand the problem it solves. Could it be possible that the medicine is worse than the sickness here?

Reply:
As I noted above to a similar comment by Janne, this was from when I was experimenting with turning on https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes (which I ended up not doing).

I will go through and remove these explicit `| undefined`

Reply:
>I wonder if this specific new kind of strictness in typing is worth it. I do not understand the problem it solves. Could it be possible that the medicine is worse than the sickness here?

This is not answered :/ Are you sure this strictness is not adding complexity without return?

Reply:
That's not what I think, and I haven't made up my mind. It is enough for me if you indeed are sure about this. I ask because sometimes people do stuff like this on auto-pilot and the common good requires a bit of second-guessing.

Another view for thought: feeling confident about this subset of bugs could be the byproduct of unit testing. But we don't have those yet. I think that's the root of the problem.

Resolved.